solr/core/src/java/org/apache/solr/security/JWTAuthPlugin.java

```diff
@@ -155,8 +168,33 @@ public void init(Map<String, Object> pluginConfig) {
       requiredScopes = Arrays.asList(requiredScopesStr.split("\\s+"));
     }
 
+    // Parse custom IDP SSL Cert from either path or string
+    InputStream trustedCertsStream = null;
+    String trustedCertsFile = (String) pluginConfig.get(PARAM_TRUSTED_CERTS_FILE);
+    String trustedCerts = (String) pluginConfig.get(PARAM_TRUSTED_CERTS);
+    if (trustedCertsFile != null && trustedCerts != null) {
+      throw new SolrException(SolrException.ErrorCode.SERVER_ERROR, "Found both " + PARAM_TRUSTED_CERTS_FILE + " and " + PARAM_TRUSTED_CERTS + ", please use only one");
+    }
+    if (trustedCertsFile != null) {
+      try {
+          trustedCertsStream = Files.newInputStream(Paths.get(trustedCertsFile));
```

Comment:
Hmm, I tried using `CoreContainer.assertPathAllowed()` but unfortunately AuthPlugins do not have access to coreContainer, so I cannot get access to it. An Auth plugin is at the container level, not the core level, so what is the equivalent of `SolrCoreAware` for such plugins?  I.e. that it would get a reference to `CoreContainer`?

Reply:
CoreContainer:517 attempts to create an authentication plugin using a constructor with the CoreContainer as a parameter first, then falls back to a zero-arg constructor. I think you can just add this constructor and hold a reference to the container.

Reply:
Thanks a lot - it actually worked!!

Solr should probably replace its dependency injection system at some point :)